op-service/tasks/poll_test.go

```diff
@@ -0,0 +1,68 @@
+package tasks
+
+import (
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+)
+
+func TestPoller(t *testing.T) {
+	cl := clock.NewDeterministicClock(time.Now())
+	counter := new(atomic.Int64)
+	poller := NewPoller(func() {
+		counter.Add(1)
+	}, cl, time.Second*5)
+
+	poller.Start()
+
+	cl.AdvanceTime(time.Second * 6) // hit the first tick
+
+	require.Eventually(t, func() bool {
+		t.Log("counter", counter.Load())
+		return counter.Load() == 1
+	}, time.Second, time.Millisecond*100)
```

Comment:
100ms isn't going to be long enough given CI servers are often overloaded.  Since this should pass as soon as the condition is reached I'd make it more like 10 seconds - it will only actually take that long when failing, but it gives plenty of head room for slow CI to avoid flakes.

Same thing for the other timeouts in this test.